lib/statsd.rb

```diff
@@ -127,7 +144,80 @@ def send_method
     def is_localhost?
       @host == 'localhost' || @host == '127.0.0.1'
     end
+  end
 
+  # Some more unabashed borrowing: https://github.com/reinh/statsd/blob/master/lib/statsd.rb#L410
+  # The big difference between this implementation and reinh's is that we don't support namespaces,
+  # and we have a bunch of hacks for introducing prefixes to the namespaces we're acting against.
+  # = Batch: A batching statsd proxy
+  #
+  # @example Batch a set of instruments using Batch and manual flush:
+  #   $statsd = Statsd.new 'localhost', 8125
+  #   batch = Statsd::Batch.new($statsd)
+  #   batch.increment 'garets'
+  #   batch.timing 'glork', 320
+  #   batch.gauge 'bork', 100
+  #   batch.flush
+  #
+  # Batch is a subclass of Statsd, but with a constructor that proxies to a
+  # normal Statsd instance. It has it's own batch_size parameters
+  # (that inherit defaults from the supplied Statsd instance). It is recommended
+  # that some care is taken if setting very large batch sizes. If the batch size
+  # exceeds the allowed packet size for UDP on your network, communication
+  # troubles may occur and data will be lost.
```

Comment:
If this happens, how will anyone know? I'm really not fond of adding mysterious failure cases.
Can we determine if the batch size is too big at Batch instantiation time?